docs/tutorial.md

```diff
@@ -737,3 +731,90 @@ Note that even though adding type annotations to .lua files makes it invalid
 Lua, you can still do so and load them from the Lua VM once the Teal package
 loader is installed by calling tl.loader().
 
+
+### Type definitions for third party libraries
+
+You can also create declaration files to annotate the types of third-party Lua
+modules, including C Lua modules. To do so, create a file with the .d.tl
+extension and require it as normal, i.e. `require("minetest")`.
+
+Types defined in this module will will be used as a source of type information 
+checking with `tl check`, even though the real Lua module will be loaded
+instead when requiring the module from Lua or `tl run`.
+
+If the third party library is "globally available", i.e. you do not explicitly
+`require` it, you can inject the types without affecting your generated code
+with a `tlconfig.lua`:
+
+```lua
+return {
+    preload_modules = {
+        "minetest" -- Execute the equivalent of require('minetest') before type-checking the tl script(s)
+    }
+}
+```
+
+#### Visibility
+
+It should be noted that local references in type definition files (or any files
+containing types for that matter) are not available in the requiring module.
+
+This means that constructs like:
+
+```lua
+local MyPointType = record
+    x :number
```

Comment:
Could you please keep the consistency of where the `:` is placed?
In the rest of the file, the colon is before the space, like `var: type` and in most of the code you added you place it after the space, like `var :type`.

I think it should be consistent in the same file, so change the added code to use `var: type`